nixos/tests/systemd-confinement.nix

```diff
@@ -80,6 +86,15 @@ import ./make-test-python.nix {
               machine.fail("chroot-exec touch /bin/test")
         '';
       }
+      { config.serviceConfig.DynamicUser = true;
```

Comment:
I'd also do this same test for the `full-apivfs` and `chroot-only` modes, since we really want to make sure that `DynamicUser` behaves the same. One difference when using `chroot-only` is for example that there is a `/var/tmp` directory when using `DynamicUser`.